src/map.c

```diff
@@ -664,6 +664,49 @@ int wally_map_keypath_add(struct wally_map *map_in,
     return ret;
 }
 
+int wally_map_tap_keypath_add(struct wally_map *hashes_map_in,
+                          struct wally_map *paths_map_in,
+                          const unsigned char *pub_key, size_t pub_key_len,
+                          const unsigned char *tapleaf_hashes, size_t tapleaf_hashes_len,
+                          const unsigned char *fingerprint, size_t fingerprint_len,
+                          const uint32_t *path, size_t path_len)
```

Comment:
At the `wally_map` interface level, I think its better to split this into two functions, and remove `tap` from the name for potential future re-use of maps holding merkle paths.

IIUC though, the final part of this impl is just `wally_map_keypath_add` on `paths_map_in`. I'd leave that as a separate call since its already implemented/exposed.

This function then would become `wally_map_merkle_path add` which just adds the merkle path keyed by the pubkey to `hashes_map_in`. If `hashes_map_in` has a verify function that only allows values from (0..N) * SHA256_LEN in length (which for psbt it does, see `map_leaf_hashes_verify`) then I think it also becomes redundant at the `wally_map` interface level.

For psbt in contrast, since (1) you want to add entries to both maps in tandem, and (2) this is taproot specific then implementing `wally_psbt_[input|output]_taproot_path_add` taking both the hashes and path info makes sense.

Reply:
> IIUC though, the final part of this impl is just wally_map_keypath_add on paths_map_in. I'd leave that as a separate call since its already implemented/exposed.

The key(badumsh) difference here is that `wally_keypath_public_key_verify` doesn't allow xonly keys. Should I relax `wally_map_keypath_add` to accept `wally_keypath_xonly_public_key_verify` as a `verify_fn` for the map, then reuse it?